traffic_ops/app/lib/API/DeliveryServiceRegexes.pm

```diff
@@ -222,6 +262,18 @@ sub delete {
 		return $self->forbidden();
 	}
 
+	my $ds = $self->db->resultset('Deliveryservice')->find( { id => $ds_id } );
+	if ( !defined($ds) ) {
+		#allow deletion if the ds is not valid
```

Comment:
can  you make this change? this doesn't make sense to me. if the ds doesn't exist, don't worry about deleting ds regexes...just throw a not_found()